core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcPreparedStatementUtils.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.jdbc;
+
+import org.apache.calcite.DataContext;
+
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Date;
+import java.sql.NClob;
+import java.sql.PreparedStatement;
+import java.sql.Ref;
+import java.sql.RowId;
+import java.sql.SQLException;
+import java.sql.SQLXML;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.function.Consumer;
+
+/**
+ * Utils for handling prepared statement decoration, like setting dynamic parameters
+ */
+public class JdbcPreparedStatementUtils {
+
+  private JdbcPreparedStatementUtils() {
+  }
+
+  public static Consumer<PreparedStatement> createPreparedStatementConsumer(Integer[] indexes,
```

Comment:
I wonder if this method should receive `List<RelDataType>`? (e.g. for proper null binding)

Reply:
Please, elaborate more on this. I think we should follow intention of the end user. If some set on the user code

`pstm.setInt(3, 123);`

then we should use the exactly the same method:

`jdbcstm.setInt(shifted-index, 123`

Reply:
I think my comment is no longer valid, and type handling should be probably be delegated to `TypedValue` (a comment above